### PR TITLE
Core/Scripts: NPC Iruk gossip from DB

### PR DESCRIPTION
**Pull request by poszer**
NPC Iruk use gossip from DB [Iruk](https://www.wowhead.com/npc=26219/iruk)
[Quest for it](https://www.wowhead.com/quest=11961)

##### CHANGES PROPOSED:

-  Added already existing DB gossip content.
-  Added correct spell enum from spell_creature_ to spell_create_.
-  Replaced switch case with if statement.

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

========== Build: 3 succeeded, 0 failed, 12 up-to-date, 1 skipped ==========
Builded without errors on Windows. Tested in game, everything work as intended, quest can be done normally, the interaction with the Iruk is also okay, the gossip works.
Picture below, with this commit : 

IrukFix

Infos : 
Wowhead

##### HOW TO TEST THE CHANGES:

1. .quest add 11961
2. .go creature 105531  - Try to speak with Iruk and click on the menu. ( You should be able to finish quest ).
[Quest Video](https://www.youtube.com/watch?v=ty36oy51Rks)

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 1]
```diff
@@ -307,13 +307,15 @@ class npc_corastrasza : public CreatureScript
 ## npc_iruk
 ######*/
 
-#define GOSSIP_ITEM_I  "<Search corpse for Issliruk's Totem.>"
-
 enum Iruk
 {
+	GOSSIP_MENU_ID_NPC_IRUK                 = 9280,
+	GOSSIP_OPTION_SEARCH_CORPSE             = 0,
+	PC_TEXT_THIS_YOUNG_TUSKARR              = 12585,
```
Comment (Nefertumm) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp]: Did you mean `NPC_TEXT_THIS_YOUNG_TUSKARR` ?

Comment (wizzymore) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: yep, it misses the N

**Comment by wizzymore** [comment]
It works as intended, great job @poszer

File: src/server/scripts/Northrend/zone_borean_tundra.cpp (hunk 1 above)
Comment (poszer) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: yeah, thanks. It's cp from my old commit :+1:

**Comment by Nefertumm** [comment]
Resolve the merge conflicts.

**Comment by NixAJ** [comment]
> 
> 
> Resolve the merge conflicts.

Can you elaborate on the merge conflicts?

**Comment by poszer** [comment]
> Resolve the merge conflicts.

dunno why :thinking: It more looks like Git issue idk

**Comment by FrancescoBorzi** [comment]
image

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 2]
```diff
@@ -307,13 +307,15 @@ class npc_corastrasza : public CreatureScript
 ## npc_iruk
 ######*/
 
-#define GOSSIP_ITEM_I  "<Search corpse for Issliruk's Totem.>"
-
 enum Iruk
 {
+	GOSSIP_MENU_ID_NPC_IRUK                 = 9280,
+	GOSSIP_OPTION_SEARCH_CORPSE             = 0,
+	NPC_TEXT_THIS_YOUNG_TUSKARR             = 12585,
```
Comment (Nefertumm) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp]: Indentation here

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 3]
```diff
@@ -324,23 +326,23 @@ class npc_iruk : public CreatureScript
...
 
-        SendGossipMenuFor(player, GOSSIP_TEXT_I, creature);
         return true;
     }
 
     bool OnGossipSelect(Player* player, Creature* /*creature*/, uint32 /*sender*/, uint32 action) override
     {
-        ClearGossipMenuFor(player);
-        switch (action)
+        player->PlayerTalkClass->ClearMenus();
```
Comment (Nefertumm) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp]: This should be as it was.
```suggestion
        ClearGossipMenuFor(player);
```

Comment (Winfidonarleyan) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: Yes

**Pull request merged**